tests/fixtures/joint.py

```diff
@@ -37,24 +78,80 @@ class JointFixture:
     event_subscriber: KafkaEventSubscriber
     interrogator: InterrogatorPort
     kafka: KafkaFixture
+    encrypted_data: EncryptedData
 
 
 @pytest_asyncio.fixture()
 async def joint_fixture(
+    generate_keypair_fixture: KeypairFixture,  # noqa: F811
     kafka_fixture: KafkaFixture,  # noqa: F811
+    s3_fixture: S3Fixture,  # noqa: F811
 ) -> AsyncGenerator[JointFixture, None]:
     """A fixture that embeds all other fixtures for integration testing"""
     # merge configs from different sources with the default one:
-    config = get_config(sources=[kafka_fixture.config])
+
+    node_config = S3ObjectStorageNodeConfig(
+        bucket=STAGING_BUCKET_ID, credentials=s3_fixture.config
+    )
+    object_storage_config = S3ObjectStoragesConfig(
+        object_storages={S3_ENDPOINT_ALIAS: node_config}
+    )
+    config = get_config(sources=[kafka_fixture.config, object_storage_config])
+
+    await s3_fixture.populate_buckets([INBOX_BUCKET_ID, STAGING_BUCKET_ID])
 
     # Create joint_fixure using the injection
-    async with prepare_core(config=config) as interrogator:
-        async with prepare_event_subscriber(
-            config=config, interrogator_override=interrogator
-        ) as event_subscriber:
-            yield JointFixture(
-                config=config,
-                event_subscriber=event_subscriber,
-                interrogator=interrogator,
-                kafka=kafka_fixture,
-            )
+    async with prepare_core(config=config) as interrogator, prepare_event_subscriber(
+        config=config, interrogator_override=interrogator
+    ) as event_subscriber:
+        sys.set_int_max_str_digits(FILE_SIZE)
+        with big_temp_file(FILE_SIZE) as data:
+            # rewind data pointer
+            data.seek(0)
+            with tempfile.NamedTemporaryFile() as encrypted_file:
+                upload_date = datetime.utcnow().isoformat()
+                private_key = generate_keypair_fixture.private_key
+                public_key = generate_keypair_fixture.public_key
+                enc_keys = [(0, private_key, public_key)]
+
+                crypt4gh.lib.encrypt(keys=enc_keys, infile=data, outfile=encrypted_file)
+
+                # get unencrypted checksum
+                data.seek(0)
+                checksum = hashlib.sha256(data.read()).hexdigest()
+
+                encrypted_file.seek(0)
+                dec_keys = [(0, private_key, None)]
+                session_keys, _ = crypt4gh.header.deconstruct(
+                    infile=encrypted_file, keys=dec_keys, sender_pubkey=public_key
+                )
+                file_secret = session_keys[0]
+
+                offset = encrypted_file.tell()
+                # Rewind file
+                encrypted_file.seek(0)
+                obj = FileObject(
+                    file_path=Path(encrypted_file.name),
+                    bucket_id=INBOX_BUCKET_ID,
+                    object_id=OBJECT_ID,
+                )
```

Comment:
In the future, we should try to avoid relying on state population in the fixtures but rather the fixtures should offer methods to set state as needed by the tests. That makes the tests more transparent and the fixtures more versatile. Nothing to fix right now, it's already on the refactoring list for some time.